packages/main/src/plugin/container-registry.ts

```diff
@@ -290,12 +291,22 @@ export class ContainerProviderRegistry {
       internalProvider.api = undefined;
       internalProvider.libpodApi = undefined;
 
+      // if is the timeout running dont start other for events
+      // connect ENOENT \.\pipe\machine_name at PipeConnectWrap.afterConnect [as oncomplete]
+      if (this.activeTimeouts.get(containerProviderConnection.endpoint.socketPath)) {
+        return;
+      }
+
       // ok we had some errors so we need to reconnect the provider
       // delay the reconnection to avoid too many reconnections
       // retry in 5 seconds
-      setTimeout(() => {
-        this.setupConnectionAPI(internalProvider, containerProviderConnection);
-      }, this.retryDelayEvents);
+      this.activeTimeouts.set(
+        containerProviderConnection.endpoint.socketPath,
+        setTimeout(() => {
+          this.activeTimeouts.set(containerProviderConnection.endpoint.socketPath, undefined);
+          this.setupConnectionAPI(internalProvider, containerProviderConnection);
+        }, this.retryDelayEvents),
+      );
     };
 
     this.handleEvents(internalProvider.api, errorHandler);
```

Comment:
We duplicate the events everytime we call this function

https://github.com/podman-desktop/podman-desktop/blob/333f183edbebe12b50232db8dda6768d3b20eed8/packages/main/src/plugin/container-registry.ts#L148

Here `handleEvents` should return a Disposable that we should dispose before recalling it ?